### DISCUSSION TITLE
unable to connect mysql database in prisma client

### DISCUSSION THREAD

**Discussion by Mahyab** [prisma/prisma#23030] [Q&A] [answered]
Hi Prisma community  
I have a problem with mysql connection database 
the env url is

### env Url is: 
```
DATABASE_URL="mysql://root:password@localhost:3306/nextapp"
 ``` 
### the prisma Schema as simple as the documentation :
```prisma
datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
generator client {
  provider = "prisma-client-js"
}
```
I  get the error as below when 'npx prisma migrate dev'  
```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "nextapp" at "localhost:3306"

Error: P1000: Authentication failed against database server at `localhost`, the provided database credentials for `root` are not valid.

Please make sure to provide valid database credentials for the database server at `localhost`.
```
I do all the required mysql configuration and reset the mysql password but nothing changed:(
hint: i am new one in prisma so i do as documention said :)

- OS: <ubuntu>
- Database: <!MySQL,>
- Node.js version: <21.1.0>

- prisma version 
- ^5.9.1
thank you in advance :)

**Comment by ludralph** [comment]
Hi @Mahyab 👋 

Thank you for raising this question.

The error message `P1000: Authentication failed against database server at localhost, the provided database credentials for root are not valid` indicates that the issue is with the database credentials.
Here are a few things you could try:
Double-check your `DATABASE_URL` in the .env file. Make sure that the username (root) and password are correct. The format should be as follows:

mysql://USER:PASSWORD@HOST:PORT/DATABASE
In your case, it should look like this:
```
mysql://root:password@localhost:3306/nextapp
```

Ensure that the MySQL server is running and accessible at the specified host and port. Make sure that the user root has the necessary permissions to access the nextapp database on your MySQL server. Try resetting the password for the root user again, and update the DATABASE_URL accordingly. Also, ensure you do not have a `$` symbol in your password.

**Reply by Mahyab** [reply, marked as answer]
Thanks for replying
 I reset the database server  password, that doesn't work، 
After that I realized that I used special characters as #€ in mysql password 
So I've used percent encoding in database password as below 
https://stackoverflow.com/questions/63684133/prisma-cant-connect-to-postgresql

**Reply by Mileslegend** [reply]
thanks but my error is like this gitpod /workspace/Muhuandevelopers/x (master) $ npx prisma db push
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "sql12706625" at "sql12.freesqldatabase.com:3306"
Error: Unknown column 'datetime_precision' in 'field list'
   0: sql_schema_connector::flavour::mysql::connection::describe_schema
             at schema-engine/connectors/sql-schema-connector/src/flavour/mysql/connection.rs:34
   1: schema_core::commands::schema_push::Calculate `from`
             at schema-engine/core/src/commands/schema_push.rs:29
   2: schema_core::state::SchemaPush
             at schema-engine/core/src/state.rs:434 anybody who can help me out
